core/trino-parser/src/main/java/io/trino/sql/tree/SetTimeZone.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.tree;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.util.Objects.requireNonNull;
+
+public class SetTimeZone
+        extends Statement
+{
+    private final Optional<Expression> timeZone;
+
+    public SetTimeZone()
+    {
+        this(Optional.empty(), Optional.empty());
+    }
+
+    public SetTimeZone(Expression timeZone)
+    {
+        this(Optional.empty(), Optional.of(requireNonNull(timeZone, "timeZone is null")));
+    }
+
+    public SetTimeZone(NodeLocation location, Expression timeZone)
+    {
+        this(Optional.of(location), Optional.of(requireNonNull(timeZone, "timeZone is null")));
+    }
+
+    public SetTimeZone(NodeLocation location)
+    {
+        this(Optional.of(location), Optional.empty());
+    }
+
+    private SetTimeZone(Optional<NodeLocation> location, Optional<Expression> timeZone)
+    {
+        super(location);
+        this.timeZone = timeZone;
+    }
+
+    public Optional<Expression> getTimeZone()
+    {
+        return timeZone;
+    }
+
+    public boolean isLocal()
+    {
+        return !timeZone.isPresent();
```

Comment:
Use `timeZone.isEmpty()`.

Reply:
The effective `pom.xml` of `trino-parser` project is configured to work with Java source level `8`

```
<artifactId>maven-compiler-plugin</artifactId>
          <version>3.8.1</version>
          <configuration>
            <source>8</source>
            <release>8</release>
```

`Optional.isEmpty()` is available since Java 11